### PR TITLE
Use jest-environment-jsdom-sixteen

### PR DESCRIPTION
**Pull request by rickhanlonii**
## Summary
**Do not merge until JSDOM publishes a new version**

This PR replaces the [jsdom resolution override](https://github.com/facebook/react/pull/19284) with using `jest-environment-jsdom-sixteen` as the test runner until we update to [Jest 26](https://github.com/facebook/react/pull/18816)

## Test Plan

- Tests pass

### CODE REVIEW

File: scripts/jest/config.base.js [hunk 1]
```diff
@@ -21,4 +21,7 @@ module.exports = {
   collectCoverageFrom: ['packages/**/*.js'],
   timers: 'fake',
   snapshotSerializers: [require.resolve('jest-snapshot-serializer-raw')],
+
+  // TODO: Upgrade to Jest 26 which uses jsdom 16 by default.
+  testEnvironment: 'jest-environment-jsdom-sixteen',
```
Comment (SimenB) [inline scripts/jest/config.base.js:26]:
```suggestion
  testEnvironment: require.resolve('jest-environment-jsdom-sixteen'),
```

Let's Jest do less work trying to resolve it

File: yarn.lock [hunk 2]
```diff
@@ -1187,6 +1187,17 @@
...
+  resolved "https://registry.yarnpkg.com/@jest/fake-timers/-/fake-timers-25.5.0.tgz#46352e00533c024c90c2bc2ad9f2959f7f114185"
+  integrity sha512-9y2+uGnESw/oyOI3eww9yaxdZyHq7XvprfP/eeoCsjqKYts2yRlsHS/SgjPDV8FyMfn2nbMy8YzUk6nyvdLOpQ==
+  dependencies:
+    "@jest/types" "^25.5.0"
+    jest-message-util "^25.5.0"
+    jest-mock "^25.5.0"
+    jest-util "^25.5.0"
+    lolex "^5.0.0"
+
 "@jest/fake-timers@^25.2.6":
```
Comment (SimenB) [inline yarn.lock:1201]: dunno if you're up for it, but could be [deduped](https://github.com/atlassian/yarn-deduplicate) with the above

**Comment by gaearon** [comment]
We still need to wait for the jsdom release to get this in, right?

**Comment by SimenB** [comment]
https://github.com/jsdom/jsdom/pull/2996#issuecomment-656423086

File: yarn.lock (hunk 2 above)
Comment (rickhanlonii) [reply yarn.lock]: Done, also did jest-types and jest-util

**Comment by rickhanlonii** [comment]
@gaearon this should be good to go now

**Comment by rickhanlonii** [comment]
Is there a test that will fail if we don't have the JSDOM fix as expected just to double check?

**Comment by gaearon** [comment]
It's https://github.com/facebook/react/pull/19186.

**Comment by rickhanlonii** [comment]
@gaearon @trueadm I patched this onto that branch and all tests pass so I'm going to merge, let me know if there are any issues.

**Pull request merged**